src/main/java/com/hosopy/actioncable/Consumer.java

```diff
@@ -47,7 +47,10 @@ public void onFailure(Exception e) {
             @Override
             public void onMessage(String string) {
                 final Message message = Message.fromJson(string);
-                if (message.isPing()) {
+                if (message.isWelcome()) {
+                    connectionMonitor.recordConnect();
```

Comment:
@shobhitic  The same implementation exists in `Connection.Listener#onOpen()`. This leads to duplicated send of subscription command;

https://github.com/hosopy/actioncable-client-java/blob/0.1.2/src/main/java/com/hosopy/actioncable/Consumer.java#L37

```java
        this.connection.setListener(new Connection.Listener() {
            @Override
            public void onOpen() {
                connectionMonitor.recordConnect();
                subscriptions.reload();
            }
```

Removal of the implementation in `Connection.Listener#onOpen() { ... }` maybe a solution.

Reply:
Good catch. Pushed an update.